superset-frontend/packages/superset-ui-chart-controls/src/operators/pivotOperator.ts

```diff
@@ -24,12 +24,15 @@ import {
   getXAxisLabel,
 } from '@superset-ui/core';
 import { PostProcessingFactory } from './types';
+import { extractExtraMetrics } from './utils';
 
 export const pivotOperator: PostProcessingFactory<PostProcessingPivot> = (
   formData,
   queryObject,
 ) => {
   const metricLabels = ensureIsArray(queryObject.metrics).map(getMetricLabel);
+  const extraMetrics = extractExtraMetrics(formData);
+  metricLabels.push(...extraMetrics.map(getMetricLabel));
```

Comment:
a little bit code style changing in order to readability.

``` suggestion
  const metricLabels = [
    ...ensureIsArray(queryObject.metrics), 
    ...extractExtraMetrics(formData)
  ].map(getMetricLabel);
```

Reply:
Yes, much better, thanks 👍